src/function/vector_cast_operations.cpp

```diff
@@ -8,9 +8,39 @@ using namespace kuzu::common;
 namespace kuzu {
 namespace function {
 
-scalar_exec_func VectorCastOperations::bindExecFunc(
+scalar_exec_func VectorCastOperations::bindImplicitCastFunc(
     common::DataTypeID sourceTypeID, common::DataTypeID targetTypeID) {
     switch (sourceTypeID) {
+    case common::INT16: {
+        switch (targetTypeID) {
+        case common::INT32:
```

Comment:
Now you might want to warp the second switch case as its own function.

So you will have
```
castINT16(targetTypeID)
castINTXX()
```

Reply:
done